File: backend_dotnet/fruit_api/fruit_api/Models/Category.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Categories")]
public class Category
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("categoryId")]
    public string CategoryId { get; set; } = null!;

    [Required]
    [MaxLength(100)]
    [Column("categoryName")]
    public string CategoryName { get; set; } = string.Empty;

    [MaxLength(255)]
    [Column("description")]
    public string? Description { get; set; }

    [Column("imageUrl")]
    [MaxLength(255)]
    public string? ImageUrl { get; set; }

    [Column("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    // Navigation properties
    public ICollection<Product>? Products { get; set; }
}

File: frontend_flutter/fruit_market/lib/models/Category.dart
import 'product.dart';

class Category {
  final String categoryId;
  final String categoryName;
  final String? description;
  final String? imageUrl;
  final DateTime createdAt;
  final int productCount;

  /// nếu API trả luôn danh sách sản phẩm (có thể có hoặc không)
  final List<Product>? products;

  Category({
    required this.categoryId,
    required this.categoryName,
    this.description,
    this.imageUrl,
    required this.createdAt,
    required this.productCount,
    this.products,
  });

  /// JSON -> Object
  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      categoryId: json['categoryId'] ?? '',
      categoryName: json['categoryName'] ?? '',
      description: json['description'],
      imageUrl: json['imageUrl'],           // THÊM: lấy imageUrl
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
      productCount: json['productCount'] ?? 0, // THÊM: lấy productCount
      products: json['products'] != null
          ? (json['products'] as List)
              .map((e) => Product.fromJson(e))
              .toList()
          : null,
    );
  }

  /// Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'categoryId': categoryId,
      'categoryName': categoryName,
      'description': description,
      'imageUrl': imageUrl,                   // THÊM imageUrl
      'createdAt': createdAt.toIso8601String(),
      'productCount': productCount,            // THÊM productCount
      'products': products?.map((e) => e.toJson()).toList(),
    };
  }
}